File: backend/battle/src/main/java/com/nimuairy/battle/core/Player.java
package com.nimuairy.battle.core;

import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.Set;

@Getter
@Setter
public abstract class Player {

	private String name;
	private Orb orb;
	private Wall wall;
	private Set<FieldCharacter> fieldCharacters;

	Player() {

	}

	Player(String name) {
		this.name = name;
	}

//	public List<WallCharacter> getWallCharacters() {
//		return wall.getHero();
//	}
}

File: frontend/src/app/battle/models/player.ts
import { PlayerSide } from '../battle/enums/player-side';
import { FieldCharacter } from './field-character';
import { Orb } from './orb';
import { Wall } from './wall';


export interface Player {
  name: string;
  orb: Orb;
  wall: Wall;
  fieldCharacters: FieldCharacter[];
  side?: PlayerSide;
}
